src/ccl_background.c

```diff
@@ -1013,6 +1013,68 @@ void ccl_comoving_angular_distances(ccl_cosmology * cosmo, int na, double a[],
   }
 }
 
+
+double ccl_comoving_angular_diameter_distance(ccl_cosmology * cosmo, double a1, double a2, int* status)
+{
+  if(a1>1. || a2>1.) {
+    *status = CCL_ERROR_COMPUTECHI;
+    ccl_raise_warning(*status,"CCL_ERROR_COMPUTECHI: scale factor cannot be larger than 1.");
+    return NAN;
+  } else {
+    if(cosmo->params.Omega_k<0.){
+      *status = CCL_ERROR_COMPUTECHI;
+      ccl_raise_warning(*status,"CCL_ERROR_COMPUTECHI: Omega_k cannot be negative for angular diameter distance.");
```

Comment:
IDK what `ccl_raise_warning` does. Seems better to just set the cosmology status message. This will get raised by python.

Reply:
Addressed